examples/vision/fixres.py

```diff
@@ -0,0 +1,319 @@
+As shown by Touvron et al., if we can fix this discrepency then we can significantly
+boost model performance. 
+
+In this example, we will implement the techniques introduced by Touvron et al. in this
+regard. As a note, these techniques are collectively referred to as **FixRes**.
+"""
+
+"""
+## Imports
+"""
+
+from tensorflow import keras
+from tensorflow.keras import layers
+import tensorflow as tf
+
+import tensorflow_datasets as tfds
+
+tfds.disable_progress_bar()
+
+import matplotlib.pyplot as plt
+
+"""
+## Load the `tf_flowers` dataset
+"""
+
+train_dataset, val_dataset = tfds.load(
+    "tf_flowers", split=["train[:90%]", "train[90%:]"], as_supervised=True
+)
+
+num_train = tf.data.experimental.cardinality(train_dataset)
+num_val = tf.data.experimental.cardinality(val_dataset)
+print(f"Number of training examples: {num_train}")
+print(f"Number of validation examples: {num_val}")
+
+"""
+## Data preprocessing utilities
+"""
+
+"""
+We will three datasets:
+
+1. A dataset with a smaller resolution - 128x128.
+2. Two dataset with a larger resolution - 224x224.
+
+Datasets with a larger resolution will differ in terms of the augmentation transforms.
+
+The idea of FixRes is to first train a model on a smaller resolution and then fine-tune
+it on the larger one. This simple yet effective recipe leads to non-trivial performance
+improvements. Please refer to the [original paper](https://arxiv.org/abs/1906.06423) for
+results.
+"""
+
+# Reference: https://github.com/facebookresearch/FixRes/blob/main/transforms_v2.py.
+
+batch_size = 128
+auto = tf.data.AUTOTUNE
+smaller_size = 128
+bigger_size = 224
+
+size_for_resizing = int((bigger_size / smaller_size) * bigger_size)
+central_crop_layer = layers.CenterCrop(bigger_size, bigger_size)
+
+
+def make_dataset(
+    dataset: tf.data.Dataset,
+    train: bool,
+    image_size: int = smaller_size,
+    fixres: bool = True,
+):
+    def preprocess_initial(image, label):
+        """
+        Initial preprocessing function for training on smaller resolution.
+        For training, do random_horizontal_flip -> random_crop.
+        For validation, just resize.
+        No color-jittering has been used.
+        """
+        if train:
+            channels = image.shape[-1]
+            begin, size, _ = tf.image.sample_distorted_bounding_box(
+                tf.shape(image),
+                tf.zeros([0, 0, 4], tf.float32),
+                area_range=(0.05, 1.0),
+                min_object_covered=0,
+                use_image_if_no_bounding_boxes=True,
+            )
+            image = tf.slice(image, begin, size)
+
+            image.set_shape([None, None, channels])
+            image = tf.image.resize(image, [image_size, image_size])
+            image = tf.image.random_flip_left_right(image)
+        else:
+            image = tf.image.resize(image, [image_size, image_size])
+
+        image = (image - 127.5) / 127.5
```

Comment:
This line is used in 2 different places, and any user of the trained model would have to remember to apply it to their data. Perhaps you could instead put a `Rescaling` layer directly in the model.